### ISSUE TITLE
distribute less pre-trained HMMs: evaluate seqIdentity parameter in getKEGGModelForOrganism

### ISSUE THREAD

**Issue by edkerk** [SysBioChalmers/RAVEN#394]
### Description of the issue:

**tldr; we'll only keep distributing pretrained HMMs where the sequence identity treshold was set at 90%.**

Since RAVEN release 1.9.1 (commit dd7e2e758fe011253c3d192bd38d840c5747821d), CD-HIT is implemented to reduce the KEGG-provided sequences to non-redundant sets by clustering similar sequences, prior to multi-sequence alignment by MAFFT and HMM training by HMMer. Since then, we have been distributing 3 sets of pre-trained HMMs for each group (eukaryote & prokaryote), namely where CD-HIT was run with its sequence identity threshold set at either 100%; 90% or 50% (where 90% is the default setting of CD-HIT).

However, the effect of this was never tested. Meanwhile, increasing numbers of sequences being included in KEGG makes it more and more time-consuming to prepare multiple sets of pre-trained HMMs. It would be preferred if we can just distribute one set of pretrained HMMs per group (so only one for eukaryote and one for prokaryote). From this, two questions arise: (1) is there much difference between the three threshold levels when generating a model; (2) is any of the three thresholds better and should be selected for future HMMs?

### Approach
We will train HMMs with the latest KEGG version (100) with the three above mentioned CD-HIT thresholds, for both prokaryote and eukaryote. Then, we will generate models for _S. cerevisiae_, _E. coli_ and _H. sapiens_, with the argument that these model organisms are well-curated on KEGG: we can compare the HMM-derived models with models that were obtained by only providing the KEGG id. Other parameters can also affect model content, but are kept at default values:

```matlab
prok50=getKEGGModelForOrganism('eco','eco.fa','prok50_kegg100','out_prok50_kegg100',true,true,true,true,10^-50,0.8,0.3,-1,inf,0.5);
prok90=getKEGGModelForOrganism('eco','eco.fa','prok90_kegg100','out_prok90_kegg100',true,true,true,true,10^-50,0.8,0.3,-1,inf,0.9);
prok100=getKEGGModelForOrganism('eco','eco.fa','prok100_kegg100','out_prok100_kegg100',true,true,true,true,10^-50,0.8,0.3,-1,inf,1);
keggProk=getKEGGModelForOrganism('eco');
```
(and similarly for _S. cerevisiae_ and _H. sapiens_)

The KEGG-id-only model was set as gold-standard for comparison. The resulting HMMs models were then scored by: (1) number of reactions that were missing relative to the gold-standard; (2) number of unique reactions that were only observed in _this_ HMM-based model. Both numbers should be as low as possible, as we assume that KEGG is well curated for these organisms and should therefore not have reactions that should be missing (i.e. (1)), nor are missing reactions that should be present (i.e. (2)). The list of included reactions were also compared across models, but there is no reliable way to judge whether reactions were correct or not. Therefore we just sticked to using the KEGG-id-only model as gold-standard.

### Results

| Species | HMM dataset | Missing reactions (relative to KEGG annotation) | Unique reactions in _this_ HMM-based model | 
|-|-|-|-|
| _S. cerevisiae_ | euk100 | 54 | 5 |
| _S. cerevisiae_ | euk90 | 47 | 2 |
| _S. cerevisiae_ | euk50 | 49 | 29 |
| _H. sapiens_ | euk100 | 0 | 3 |
| _H. sapiens_ | euk90 | 0 | 3 |
| _H. sapiens_ | euk50 | 10 | 21 |
| _E. coli_ | prok100 | 42 | 9 |
| _E. coli_ | prok90 | 45 | 0 |
| _E. coli_ | prok50 | 56 | 30 |

### Conclusion
Models derived from 50% cutoff HMMs performed worst, both missing the most reactions that should be there and including the most unique reactions that should not be there. Between 90% and 100% was little difference. One can argue that the effect of (2) is slightly more severe (including reactions that should not be present), as (1) can always be resolved by gap-filling if the missing reaction was essential. Therefore, the 90% cutoff can be identified as best-performing (it had 2 and 0 unique reactions in _S. cerevisiae_ and _E. coli_).

#### System information
1. RAVEN version: from `chore/updateKEGG` branch, commit a8dc1d815fdf4bb45026eb0f9950c97451c206aa.
2. Operating system: Windows 10

**I hereby confirm that I have:**
- [x] Followed the [guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/Installation) to install RAVEN.
- [x] Checked that a similar issue does not [already exist](https://github.com/SysBioChalmers/RAVEN/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [ ] If suitable, needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about the issue

**Comment by simas232** [comment]
Thank you for this detailed analysis, @edkerk. I see that in terms of accuracy, the 90% cutoff provided the best results. If one aims to obtain the draft model for the non-KEGG species, it makes perfect sense to use KEGG HMM sets that were trained upon 90% cutoff. Meanwhile, for the species that are enlisted in KEGG, it may seem more appropriate to combine FASTAless reconstruction and the 50% cutoff HMM search (and 90%, 100%, if their unique reactions complement the previous searches).

**Comment by edkerk** [comment]
Thanks Simonas. I however don't agree with your second statement (for KEGG-enlisted species).

I take the assumption that _E. coli_, _S. cerevisiae_ and _H. sapiens_ are well annotated in KEGG. So I would not expect that RAVEN's HMM-based search suddenly finds 20-30 new reactions that were previously not annotated for these well-curated organisms. I could of course in-depth investigate whether these 80 new reactions are feasible, but I won't.... But a quick check of them suggests that these _new reactions_ are typically for genes that were already in the model, but now with a different substrate. These "new" substrates are often not correct when I quickly check literature. Meanwhile, versions of those reactions with the correct substrates are already included in all models.

So 50% HMM seems to be slightly less strict on the substrate specificity for some reactions, but again, this was not extremely thoroughly investigated. I still lean towards only distributing 90% HMMs versions.
